src/gui/wizard/owncloudwizard.cpp

```diff
@@ -136,7 +136,9 @@ void OwncloudWizard::adjustWizardSize()
     });
 
     Q_ASSERT(maxPageSizeIter != pageSizes.end());
-    resize(*maxPageSizeIter);
+    const auto maxPageSizeWidest = maxPageSizeIter->width() > maxPageSizeIter->height() ? maxPageSizeIter->width() : maxPageSizeIter->height();
+    const QSize newWizardSize(maxPageSizeWidest, maxPageSizeWidest);
+    resize(newWizardSize);
```

Comment:
I'd argue this is still incorrect. I get it you aim for a square but that might not be always the right one for the reasons I mentioned before, should have been std::accumulate and not std::max_element, my apologies for sending you astray there.

Reply:
I must admit that I do not understand how I can `std::accumulate` here, because it sums up the elements and I want to find out what's the longest element.

I also would argue that the logic is correct because after `std::max_element` I come out with the page size that has the longest side in it. Then I check again if the longest side is the width or the height of the size element. Then I would think I have figured out what the longest side is.

Reply:
> I must admit that I do not understand how I can use `std::accumulate` here, because it sums up the elements and I want to find out what's the longest element.

I guess it's the part I don't get. But in any case it doesn't "sum" per say, it "accumulates" and you control what is the initial value and how to compute the accumulation. I'd think something along those lines is what you'd want:

```
const auto maxPageSize = std::accumulate(pageSizes.begin(), pageSizes.end(), QSize(), [](const QSize &a, const QSize &b) {
        return QSize(qMax(a.width(), b.width()), qMax(a.height(), b.height()));
    });
```

> I also would argue that the logic is correct because after `std::max_element` I come out with the page size that has the longest side in it. Then I check again if the longest side is the width or the height of the size element. Then I would think I have figured out what the longest side is.

Right, probably me who misread, this lacks a few use of qMax/std::max instead of ternary operator in there. ;-)

Reply:
I think most of the discussion here still applies.

Reply:
Ok I changed the code to use std::accumulate.